src/swarm-tui/Swarm/TUI/Model.hs

```diff
@@ -139,6 +146,24 @@ logEvent src sev who msg el =
  where
   l = LogEntry (TickNumber 0) src sev who msg
 
+data KeyEventHandlingState = KeyEventHandlingState
+  { _keyConfig :: KeyConfig SwarmEvent
+  , _keyHandlers :: EventHandlers
+  }
+
+keyConfig :: Lens' KeyEventHandlingState (KeyConfig SwarmEvent)
```

Comment:
Should be possible to just `makeLenses ''KeyEventHandlingState`, I don't think these need to be written by hand.

Reply:
It is not because of the TH splice `KeyEventHandlingState` would not see `AppState`.

I could move the lenses to the end of file or some other trick, I guess...

Reply:
Ah, yeah, sometimes I have to reorder things to make the staging work out.  Defining these by hand is also  fine, they are pretty simple.  Maybe just put a comment explaining why they have to be defined by hand.

Reply:
I finally moved them to the end of the file.